### ISSUE TITLE
Exception: Number of retries reached 0.

### ISSUE THREAD

**Issue by andr-04** [LonamiWebs/Telethon#270]
I haven't check on new version v0.14, but previous one this exception happened very often on outgoing requests in a random time. I don't know: Telegram servers sometime is slow or it's flood protection. Definitely this exception has to be more informative to understand the reason of it.

**Comment by Lonami** [comment]
I any request has a `None` result:
https://github.com/LonamiWebs/Telethon/blob/3fda7efeb9b1682197ed48f519c0e3673d350ce7/telethon/telegram_bare_client.py#L345

It's going to retry:
https://github.com/LonamiWebs/Telethon/blob/3fda7efeb9b1682197ed48f519c0e3673d350ce7/telethon/telegram_bare_client.py#L348-L350

Up to five times before raising an error:
https://github.com/LonamiWebs/Telethon/blob/3fda7efeb9b1682197ed48f519c0e3673d350ce7/telethon/telegram_bare_client.py#L307-L308

I don't know why the result of some requests `is None`. I know it fails for `LogOutRequest`, need to fix that, forgot to do it before `v0.14.1`.

**Comment by andr-04** [comment]
Maybe the exception on `LogOutRequest` is related with this as well?

**Comment by Lonami** [comment]
Yes, the exception there is this one, because logging out is "confirmed" through a message acknowledge and no result is actually read. From my experience, I think ack requests are only used int this case… So no result per se is set.

**Comment by Lonami** [comment]
Seems to work with `LogOutRequest` but I'm yet to decide what to do with every other request. Maybe reconnect? But what request is it the one which fails? Is it one in specific? Maybe no result is returned ever? Or does it happen with all?

**Comment by andr-04** [comment]
In logs I see the problem only with `ResolveUsernameRequest`. Probably other requests have this problem as well, but my use case raises the exception only with this one.

**Comment by Lonami** [comment]
There was this slightly obscure bug I just solved on the branch `many-connections`, where the result was also `None`. Turned out the client was *not* connected, so it retried all five times really quickly, and finally returned `None`… somehow. Need to investigate more.

**Comment by Lonami** [comment]
The error may be because `.disconnect()` was called from somewhere in the code, and it's trying to invoke while *not* being connected. Thus it fails and doesn't attempt to reconnect. Need to find where though.

**Comment by 88ee55** [comment]
I have the same exception.
channels.ReadHistoryRequest(channel=peer, max_id=0)
peer is InputChannel

telethon 0.15.0.3
python 3.5

**Comment by 88ee55** [comment]
ReadHistoryRequest
You need to call if unread more than 0
then there will be no exception

**Comment by Lonami** [comment]
So are you saying that with `max_id=0` it fails all retrues, but works with a non-zero value? I wonder what Telegram is replying or doing for this to happen with only certain parameters.

**Comment by 88ee55** [comment]
exception occurs when there are no unread messages

**Comment by Lonami** [comment]
If anyone wants to test, this would be really helpful doing this would be really helpful. Add this class to `telethon/tl/tlobject.py`:
```python
from threading import Event
import traceback
import os
class LoggingEvent:
    def __init__(self, t):
        self.o = open(os.path.expanduser('~/telethon.log'), 'w')
        self.e = Event()
        self.t = type(t).__name__
    def _log(self, kind):
        msg = '-- '+kind+' FLAG ON '+self.t.+' --'
        self.o.write(msg+'\n')
        traceback.print_stack(file=self.o)
        self.o.write('-'*len(msg)+'\n\n\n')
        self.o.flush()
    def clear(self):
        self._log('CLEARING')
        return self.e.clear()
    def set(self):
        self._log('SETTING')
        return self.e.set()
    def is_set(self):
        return self.e.is_set()
    def wait(self, *a, **k):
        self.e.wait(*a, **k)
```

Then, inside the `TLObject` class change:
```python
self.confirm_received = Event()
```

with
```python
self.confirm_received = LoggingEvent(self)
```

**Comment by alihossein** [comment]
> exception occurs when there are no unread messages
how can I check a number of unread messages?

**Comment by 88ee55** [comment]
> how can I check the number of unread messages?
`GetFullChannelRequest(InputChannel()).chat_full.unread_count`

**Comment by Lonami** [comment]
@alihossein this isn't the place to ask. Keep the issue on topic or open a new issue. Or read the manual.

**Comment by Lonami** [comment]
> exception occurs when there are no unread messages

This is false-y, so it's probably the reason why the exception occured. Can you test again?

**Comment by mnvx** [comment]
I has stuck on this Exception too. Currently I skip these requests:

```python
    def my_method(self):
        try:
            dialogs, entities = self._client.get_dialogs()
        except Exception as e:
            time.sleep(1) # I hope what after delay connection will be repaired
            raise MyApiException('Im sorry, try later')
        return dialogs

    def call_my_method(self):
        try:
            dialogs = self.my_method()
        except MyApiException as e:
            self._log(str(e))
            return

        self.process_dialogs(dialogs)
```

**Comment by K900** [comment]
I've reproduced the bug successfully. Here's the log:
```
-- SETTING FLAG ON TLMessage --
  File ".../dump_tg.py", line 38, in <module>
    main()
  File ".../dump_tg.py", line 32, in main
    download_chat(client, b, 'b')
  File ".../dump_tg.py", line 23, in download_chat
    downloader.download_from_message(message)
  File "...\downloader.py", line 92, in download_from_message
    output_path=path
  File "...\downloader.py", line 56, in download_if_needed
    progress_callback=progress_callback
  File "C:\Users\...\AppData\Local\Programs\Python\Python36-32\lib\site-packages\telethon\telegram_bare_client.py", line 701, in download_file
    input_location, offset, part_size
  File "C:\Users\...\AppData\Local\Programs\Python\Python36-32\lib\site-packages\telethon\telegram_bare_client.py", line 448, in __call__
    result = self._invoke(sender, call_receive, *requests)
  File "C:\Users\...\AppData\Local\Programs\Python\Python36-32\lib\site-packages\telethon\telegram_bare_client.py", line 503, in _invoke
    sender.disconnect()
  File "C:\Users\...\AppData\Local\Programs\Python\Python36-32\lib\site-packages\telethon\network\mtproto_sender.py", line 55, in disconnect
    self._clear_all_pending()
  File "C:\Users\k0009\AppData\Local\Programs\Python\Python36-32\lib\site-packages\telethon\network\mtproto_sender.py", line 247, in _clear_all_pending
    r.confirm_received.set()
  File "C:\Users\...\AppData\Local\Programs\Python\Python36-32\lib\site-packages\telethon\tl\tlobject.py", line 22, in set
    def set(self):
  File "C:\Users\...\AppData\Local\Programs\Python\Python36-32\lib\site-packages\telethon\tl\tlobject.py", line 15, in _log
    self.o.write(m+'\n')
-------------------------------
```

**Comment by K900** [comment]
With the latest commits from master, I now get `telethon.errors.rpc_errors_420.FloodWaitError: (FloodWaitError(...), 'A wait of 488 seconds is required.')`. Looks like the connection drop issue is fixed, but it doesn't always sleep when rate limited.

**Comment by Lonami** [comment]
No, there's a threshold (defaults to 30 seconds). If it's longer than that you don't want the library to just hang, you may want to take other actions. You should be able to modify that threshold to be as big as you need.

**Comment by K900** [comment]
Oh, I see. How do you set it up? I don't see any relevant parameters on `TelegramClient` :(

**Comment by Lonami** [comment]
@pashbelg are you willing to help or were you just confirming you have this issue too? If the latter, that's not too helpful.

**Comment by k01aty** [comment]
If I use 'InteractiveTelegramClient' instead of 'TelegramClient' .connect() ,it is not throwing this exception

**Comment by vegeta1k95** [comment]
I also get this exception randomly after doing some GetDialogsRequest. But after some pause invoking same request doesn't raise error even if dialogs did not change at all.

**Comment by andr-04** [comment]
I've fixed many aspects of this issue and the last one I've found for today.

I've found that sometimes Telegram stops send any responses. `send` returns the bytes which were sent without exceptions, but no answers from Telegram happened anymore. After some time _Errno 104: Connection reset by peer_ happened. The solution -- not just [`return None`](https://github.com/LonamiWebs/Telethon/blob/master/telethon/telegram_bare_client.py#L522), but to check if this is a retry (2nd attempt for example), make a reconnection.

**Comment by Lonami** [comment]
Is it possible to quickly check whether the socket has been closed, so we can do it when retrying? Shouldn't it error with "socket closed" while trying to read already? The background thread should be reconnecting. Maybe retries happen too quick?

**Comment by andr-04** [comment]
No, the socket is not closed because of `send` return positive number of sent bytes. As a result you just never get the result during retries. And don't detect the connection is broken -- this is the main problem.

**Comment by Lonami** [comment]
Do you think we can use [`.select`](https://docs.python.org/3/library/select.html#select.select) to determine whether reading from the socket was closed or not, like [this](https://stackoverflow.com/a/21802953/4759433) answer suggests? Or checking `if socket.fileno() == -1` as [this](https://stackoverflow.com/a/45251241/4759433) other answer suggests? I don't want to just reconnect if we are connected and the server may write.

**Comment by andr-04** [comment]
I don't know. Indeed, I've made the reconnection in my implementation, you can play. In my view you will not get an adequate results due to `send` doesn't raise exceptions. It says the socket is live.

**Comment by Lonami** [comment]
Is the `ValueError` raised immediatly or after it times out?

**Comment by andr-04** [comment]
I don't use the master version last time. I gave you a hint, you can investigate more by yourself. According the source code it can't be raised immediately, just after retries exceeded. Yes, it waits for timeout and timeout is happened.

**Comment by vegeta1k95** [comment]
Okey. I've managed to find something about this.

I had one session file and I made two copies of it.
I used FIRST copy in my script and eventually somehow managed to catch this error.
This error made my script stumble and started appearing on every requests (i.e. ResolveUsernameRequest), and even on connection request (GetStateRequest in 'sync_state' function of bare_telegram_client). I could not do anything, even multiple reconnections didn't help.

However, I started the same script in parallel with the SECOND fresh copy of the session and it started working like if nothing happened, while first was struggling to do anything. Then I replaced first copy with the second one and it also started working.

So I can assume that the cause may lie within the session file and not because of connection problems or Telegram server rejecting requests. Maybe, data, i.e. auth_key or salt, somehow becomes damaged and inflicts bad responses from servers. Unfortunately, I forgot to make a backup of "bad" copy to check if there were differences.

That's all I know at the moment. I'm trying to find the way to reproduce this, to investigate it more.

**Comment by Lonami** [comment]
Bump. Does this still happen as of `v0.16.1.x`?

**Comment by ehsanskyboy** [comment]
@Lonami yes , I have tried to forward a message in all megagroups and after about 15 times of forwarding in 2 groups, with a 5min sleep between every time, I got this error.
`  File "../modules\telethon\telegram_bare_client.py", line 459, in __call__
    raise RuntimeError('Number of retries reached 0.')
RuntimeError: Number of retries reached 0.`

**Comment by Nikita-L** [comment]
Fixed this in dirty way, but may be helpful:
```
from telethon import TelegramClient
from telethon.tl.functions.messages import SendMessageRequest


class FixedTelegramClient(TelegramClient):
    def send_message(self, entity, message, reply_to=None, parse_mode='md',
                     link_preview=True):
        entity = self.get_input_entity(entity)
        message, msg_entities = self._parse_message_text(message, parse_mode)

        request = SendMessageRequest(
            peer=entity,
            message=message,
            entities=msg_entities,
            no_webpage=not link_preview,
            reply_to_msg_id=self._get_message_id(reply_to)
        )
        result = self(request)

        return True

    def __call__(self, *requests, retries=5):
        call_receive = \
            not self._idling.is_set() or self._reconnect_lock.locked()
        for retry in range(retries):
            result = self._invoke(call_receive, *requests)
            return result


app = FixedTelegramClient('test_project', TELEGRAM_API_ID, TELEGRAM_API_HASH)
```

**Comment by rex101** [comment]
how can I create client using Telethon because when I am executing this code it says from telethon import TelegramClient

api_id = 00000
api_hash = 'aadasd6a76d7a6d'

client = TelegramClient('session_name', api_id, api_hash)
client.start()

ERROR:

Traceback (most recent call last):
  File "test.py", line 4, in <module>
    from telethon import TelegramClient
  File "/abc/bde/Desktop/Telet/Telethon/telethon/__init__.py", line 2, in <module>
    from .telegram_bare_client import TelegramBareClient
  File "/abc/bde/Desktop/Telet/Telethon/telethon/telegram_bare_client.py", line 414
    def __call__(self, *requests, retries=5):
                                        ^
SyntaxError: invalid syntax

**Comment by Lonami** [comment]
@rex101 why did you leave #644 empty and comment here‽ You probably are using Python 2 and it states literally everywhere that you **need** Python 3.

**Comment by rex101** [comment]
@Lonami Can't write it's showing " This conversation has been locked as resolved and limited to collaborators. "

**Comment by rex101** [comment]
@Lonami I solved this Syntax Error now I have new error :(
how we can create list Vector for users because I can't find any class or package in Telethon ?
That was my question and I solved it by making a Iterrable List of users but now I am getting this Error 

Traceback (most recent call last):
  File "test.py", line 86, in <module>
    r = client(InviteToChannelRequest(channel, users))
  File "C:\Users\xyz\Desktop\tel\Telethon-master\telethon\telegram_bare_client.py", line 443, in __call__
    result = self._invoke(call_receive, *requests)
  File "C:\Users\xyz\Desktop\tel\Telethon-master\telethon\telegram_bare_client.py", line 532, in _invoke
    raise next(x.rpc_error for x in requests if x.rpc_error)
telethon.errors.rpc_error_list.ChatWriteForbiddenError: You can't write in this chat 
---------------------------------------------------------------------------------------------------------

Why this is happening I am admin of that Group?

**Comment by Lonami** [comment]
@rex101 **stop commenting on completely unrelated issues**.

**Comment by bikhial** [comment]
Hi guys

I get This Number of Retries reached 0 message for GetHistoryRequest when i try to fetch channel posts. is this related to this issue or not?
if not what can i do for resolve it?
using lates version 0.18.2

thanks

**Comment by ferferga** [comment]
Same for me, but still looking for a pattern to reproduce. Will update here whenever I find something useful. However, and I don't know if it's useful or have something to do with this, when this exception is raised and I retry the exact same query, another exception with the message "Telegram's having problems right now, try again later" is raised. I use "limit=None" as others

**Comment by Lonami** [comment]
Yesterday I was having some trouble to iterate over all messages of a channel too, I got a few "internal errors from Telegram", but not "retries reached 0"

**Comment by olivier-nexmo** [comment]
I have a python script running in a loop, checking new messages every 5 minutes. After a while, the error appears:

2018-04-18 06:46:45
Error: Number of retries reached 0.
2018-04-18 06:52:36
Error: Number of retries reached 0.
2018-04-18 06:58:26
Error: Number of retries reached 0.
2018-04-18 07:04:16
Error: Number of retries reached 0.

If I kill the running script and start again, it's fine for hours. Then the error comes back.

What can I do to help to resolve this?

**Comment by Lonami** [comment]
You can create a Minimal Working Example for me to try and reproduce the issue, with the hope that I can fix it. Otherwise enable `DEBUG` logging, give me the output, and hopefully I can see something meaningful on it.

**Comment by saksham2410** [comment]
Has the issue fixed? Or anyone has found an alternative method? I get these Number of retries reached 0 errors very frequently.

**Comment by umaxfun** [comment]
Some new shiny logs:
```
[2018-05-25 11:27:50,382: INFO/MainProcess] Received task: jobs.save_channel_by_name_async[a4d161c5-8c5a-41e3-a38a-0580b5749a33]
[2018-05-25 11:27:50,399: INFO/ForkPoolWorker-2] ConnectionError "[Errno 32] Broken pipe" while writing data
[2018-05-25 11:27:50,406: WARNING/ForkPoolWorker-2] Connection was reset while invoking
[2018-05-25 11:27:50,406: INFO/ForkPoolWorker-2] Disconnecting MtProtoSender...
[2018-05-25 11:27:50,407: INFO/ForkPoolWorker-2] Abruptly confirming TLMessage
[2018-05-25 11:27:50,408: INFO/ForkPoolWorker-2] Invoking GetFullChannelRequest failed 1 times, connecting again and retrying
[2018-05-25 11:27:51,409: INFO/ForkPoolWorker-2] Attempting reconnection...
[2018-05-25 11:27:51,409: INFO/ForkPoolWorker-2] Connecting to 149.154.167.51:443...
[2018-05-25 11:27:51,466: INFO/ForkPoolWorker-2] Connection success!
[2018-05-25 11:28:01,469: WARNING/ForkPoolWorker-2] Invoking GetFullChannelRequest failed 2 times, connecting again and retrying
[2018-05-25 11:28:02,470: INFO/ForkPoolWorker-2] Reconnection aborted: already connected
[2018-05-25 11:28:12,473: WARNING/ForkPoolWorker-2] Invoking GetFullChannelRequest failed 3 times, connecting again and retrying
[2018-05-25 11:28:13,474: INFO/ForkPoolWorker-2] Reconnection aborted: already connected
[2018-05-25 11:28:23,477: WARNING/ForkPoolWorker-2] Invoking GetFullChannelRequest failed 4 times, connecting again and retrying
[2018-05-25 11:28:24,478: INFO/ForkPoolWorker-2] Reconnection aborted: already connected
[2018-05-25 11:28:34,481: WARNING/ForkPoolWorker-2] Invoking GetFullChannelRequest failed 5 times, connecting again and retrying
[2018-05-25 11:28:35,482: INFO/ForkPoolWorker-2] Reconnection aborted: already connected
[2018-05-25 11:28:35,500: ERROR/ForkPoolWorker-2] Task jobs.save_channel_by_name_async[a4d161c5-8c5a-41e3-a38a-0580b5749a33] raised unexpected: RuntimeError('Number of retries reached 0 for GetFullChannelRequest.',)
```

Seems that the problem is located here:
```
[2018-05-25 11:28:01,469: WARNING/ForkPoolWorker-2] Invoking GetFullChannelRequest failed 2 times, connecting again and retrying
[2018-05-25 11:28:02,470: INFO/ForkPoolWorker-2] Reconnection aborted: already connected
```

**Comment by Lonami** [comment]
https://github.com/LonamiWebs/Telethon/commit/bb3a5645008b01ee18f5adf1554ea7ce8a7cb15a has been merged, reopen *after* the next version is published if this error still occurs. Note that `master` is now running *under `asyncio`* so your code won't be compatible with it. To preserve this compatibility, a new branch emulating synchronous behaviour will be created.

**Comment by Lonami** [comment]
Really? `v1.0-post2`? What are the DEBUG logs? What request? How to reproduce the issue?

**Comment by endor-daniel** [comment]
@Lonami - apparently, Telegram servers were down for about an hour.

**Comment by Lonami** [comment]
I did notice that. So not an issue.

**Comment by endor-daniel** [comment]
Not an issue in Telethon, that's for sure.

**Comment by mh-iranpak** [comment]
Hi.
i have same issue while getting a channel history using `iter_messages` method. while script starts it works well. but after a while this happens. i am using `telethon 0.18.2`. And this was not happening until yesterday. this is the log for my issue:

`DEBUG:telethon.telegram_bare_client:Invoking GetHistoryRequest
DEBUG:telethon.network.mtproto_sender:Sending requests with IDs: GetHistoryRequest: 6622887426673398964
INFO:telethon.telegram_bare_client:Connection success!
DEBUG:telethon.network.mtproto_sender:Sending requests with IDs: GetStateRequest: 6622887427520293228
DEBUG:telethon.telegram_bare_client:Receiving items from the network...
DEBUG:telethon.extensions.tcp_client:socket.timeout "timed out" while reading data
DEBUG:telethon.telegram_bare_client:Receiving items from the network timed out
DEBUG:telethon.telegram_bare_client:Receiving items from the network...
DEBUG:telethon.extensions.tcp_client:socket.timeout "timed out" while reading data
WARNING:telethon.telegram_bare_client:Invoking timed out
INFO:telethon.telegram_bare_client:Invoking ['GetHistoryRequest(peer=InputPeerChannel(channel_id=1450340365, access_hash=6837884551796214524), offset_id=48, offset_date=None, add_offset=0, limit=30, max_id=0, min_id=17, hash=0)'] failed 1 times, connecting again and retrying
INFO:telethon.telegram_bare_client:Reconnection aborted: already connected
DEBUG:telethon.network.mtproto_sender:Sending requests with IDs: GetHistoryRequest: 6622887474814562696
DEBUG:telethon.extensions.tcp_client:socket.timeout "timed out" while reading data
DEBUG:telethon.telegram_bare_client:Receiving items from the network timed out
DEBUG:telethon.telegram_bare_client:Receiving items from the network...
WARNING:telethon.extensions.tcp_client:socket.timeout "timed out" when 5/8 had been received
WARNING:telethon.telegram_bare_client:Invoking timed out
WARNING:telethon.telegram_bare_client:Invoking ['GetHistoryRequest(peer=InputPeerChannel(channel_id=1450340365, access_hash=6837884551796214524), offset_id=48, offset_date=None, add_offset=0, limit=30, max_id=0, min_id=17, hash=0)'] failed 2 times, connecting again and retrying
INFO:telethon.telegram_bare_client:Reconnection aborted: already connected
DEBUG:telethon.network.mtproto_sender:Sending requests with IDs: GetHistoryRequest: 6622887522986685556
DEBUG:telethon.extensions.tcp_client:socket.timeout "timed out" while reading data
DEBUG:telethon.telegram_bare_client:Receiving items from the network timed out
DEBUG:telethon.network.mtproto_sender:Sending requests with IDs: PingRequest: 6622887556435838520
DEBUG:telethon.telegram_bare_client:Receiving items from the network...
DEBUG:telethon.extensions.tcp_client:socket.timeout "timed out" while reading data
WARNING:telethon.telegram_bare_client:Invoking timed out
WARNING:telethon.telegram_bare_client:Invoking ['GetHistoryRequest(peer=InputPeerChannel(channel_id=1450340365, access_hash=6837884551796214524), offset_id=48, offset_date=None, add_offset=0, limit=30, max_id=0, min_id=17, hash=0)'] failed 3 times, connecting again and retrying
INFO:telethon.telegram_bare_client:Reconnection aborted: already connected
DEBUG:telethon.network.mtproto_sender:Sending requests with IDs: GetHistoryRequest: 6622887570290757844
DEBUG:telethon.extensions.tcp_client:socket.timeout "timed out" while reading data
DEBUG:telethon.telegram_bare_client:Receiving items from the network timed out
DEBUG:telethon.telegram_bare_client:Receiving items from the network...
WARNING:telethon.extensions.tcp_client:socket.timeout "timed out" when 5/8 had been received
WARNING:telethon.telegram_bare_client:Invoking timed out
WARNING:telethon.telegram_bare_client:Invoking ['GetHistoryRequest(peer=InputPeerChannel(channel_id=1450340365, access_hash=6837884551796214524), offset_id=48, offset_date=None, add_offset=0, limit=30, max_id=0, min_id=17, hash=0)'] failed 4 times, connecting again and retrying
INFO:telethon.telegram_bare_client:Reconnection aborted: already connected
DEBUG:telethon.network.mtproto_sender:Sending requests with IDs: GetHistoryRequest: 6622887618782004568
DEBUG:telethon.extensions.tcp_client:socket.timeout "timed out" while reading data
DEBUG:telethon.telegram_bare_client:Receiving items from the network timed out
DEBUG:telethon.telegram_bare_client:Receiving items from the network...
DEBUG:telethon.extensions.tcp_client:socket.timeout "timed out" while reading data
WARNING:telethon.telegram_bare_client:Invoking timed out
WARNING:telethon.telegram_bare_client:Invoking ['GetHistoryRequest(peer=InputPeerChannel(channel_id=1450340365, access_hash=6837884551796214524), offset_id=48, offset_date=None, add_offset=0, limit=30, max_id=0, min_id=17, hash=0)'] failed 5 times, connecting again and retrying
INFO:telethon.telegram_bare_client:Reconnection aborted: already connected
Traceback (most recent call last):
  File "/root/vitabot/history_transfer.py", line 313, in check_channels
    channel_messages = get_channel_messages(single_channel, messsages_count)
  File "/root/vitabot/history_transfer.py", line 282, in get_channel_messages
    for message in reversed_channel_messages:
  File "/usr/local/lib/python3.6/dist-packages/telethon/telegram_client.py", line 1036, in iter_messages
    hash=0
  File "/usr/local/lib/python3.6/dist-packages/telethon/telegram_bare_client.py", line 467, in __call__
    [type(x).__name__ for x in requests]
RuntimeError: Number of retries reached 0 for ['GetHistoryRequest'].
`

Is this because of my old version of telethon?

**Comment by Lonami** [comment]
> Is this because of my old version of telethon?

Quite possibly.

**Comment by hzexe** [comment]
The same too.
>>> telethon.__version__
'1.4.1'
python 3.5

**Comment by Lonami** [comment]
It's impossible 1.4.1 shows "the same" [because that error is gone](https://github.com/LonamiWebs/Telethon/blob/2863eb8cc9f4e1e25bffeec3d1539568dd306e48/telethon/client/users.py#L88). Please show a DEBUG log with 1.4.1 if it really happens "the same".
